Add documentation on our observability stack (#1094)
giantswarm/docs: src/content/app-platform/apps/observability/_index.md

---
linkTitle: Observability Stack
title: Managed Observability Stack (Monitoring, Logging, Service Mesh)
description: Overview of the Giant Swarm managed observability stack of apps, which includes monitoring, logging, and service mesh
weight: 10
menu:
  main:
    parent: app-platform-apps
owner:
  - https://github.com/orgs/giantswarm/teams/team-halo
user_questions:
  - What app do you recommend for monitoring?
  - What app do you recommend for logging?
  - What app do you recommend for service mesh?
last_review_date: 2021-08-18
---

# Giant Swarm managed Observability Stack of Apps

> Observability is based on three main data sources: logs, metrics and tracing*. To cover these needs Giant Swarm provides its customers with a managed observability stack. The apps included in this stack are listed below. We chose these apps based on our general principles of preferring open source over proprietary solutions, as well as simplicity, performance, and speed over complexity.

Reviewer: The `*` here should be escaped at least

Reviewer: Maybe @marians has a good idea how to do this better in markdown. This leads to additional information. Maybe add a link or something? Or put the text from the botton in a `>` quote directly below this block?

> **What do we mean by a ‘managed’ app?** We mean offering similar Service Level Objectives as the rest of our offering as described in [The Radical Way Giant Swarm Handles Service Level Objectives](https://www.giantswarm.io/blog/the-radical-way-giant-swarm-handles-service-level-objectives).

Reviewer: Did you check the rendered output? These quotes look suspicious. If you wanted the word "managed" to look like this `managed`, then proper backticks should be used.

```suggestion
**What do we mean by a `managed` app?** We mean offering similar Service Level Objectives as the rest of our offering as described in [The Radical Way Giant Swarm Handles Service Level Objectives](https://www.giantswarm.io/blog/the-radical-way-giant-swarm-handles-service-level-objectives).
```

Author: thanks. will use quotations!

In brief, super fast response times from engineers, 24//7 monitoring and alerting by us, and in general, having shared goals and responsibilities to keep services up and get them up when they go down.

> In addition, we strive to uphold our “30 Day Upgrade Promise,” our commitment to keep all managed apps within 30 days of latest upstream releases.

Reviewer: Quotes should be `"` and comma inside quotes?

Author: the quote is "", yes. And in [american english](https://www.grammarly.com/blog/quotation-marks/), which we use, commas are inside quotes. What's the question? :D

Reviewer: Ok no idea about english :joy: 
but the quotes should be `"` not `“` and `”`. By using `"` we can configure hugo to render the quotes as we wish.

Author: ah ok! didn't realize oshrat used a different symbol. changed now!

TL;DR
For observability, we recommend and provide these four managed apps:

> **1/ Prometheus for monitoring / metrics**: To get all your metrics, process, store and alert on them. Prometheus has become the de-facto standard monitoring tool in the kubernetes ecosystem.

Reviewer: IMO using `##` headers here would be better than just bold text. Difference would be that bold text is percieved as headline by humans but not by computers (for toc generation for example)

Author: Ok i will change them to ##

Author: done!

**2/ Loki for logging**: To fetch logs from containers, store and index them, and make them searchable. To install, go to [Loki README](https://github.com/giantswarm/loki-app/blob/master/README.md).

> In addition, we understand some customers already know and use Elasticsearch. For this reason, we also provide a managed EFK stack (Elasticsearch, Fluentd, and Kibana) for logging. To install, go to [EFK Stack README](https://github.com/giantswarm/efk-stack-app).

Reviewer: The link strays from the other links, either change this one to link directly to the readme or change the other ones to point to just the repo.

Author: ok! changing this one to readme.

---

> linkTitle: Observability Stack
> title: Managed Observability Stack (Monitoring, Logging, Service Mesh)

Reviewer 2: ```suggestion
linkTitle: Observability stack
title: Managed observability stack (monitoring, logging, service mesh)
```

FYI: We decided for sentence casing instead of title case.

description: Overview of the Giant Swarm managed observability stack of apps, which includes monitoring, logging, and service mesh
weight: 10
menu:
  main:
    parent: app-platform-apps
owner:
  - https://github.com/orgs/giantswarm/teams/team-halo
user_questions:
  - What app do you recommend for monitoring?
  - What app do you recommend for logging?
  - What app do you recommend for service mesh?
last_review_date: 2021-08-18
---

> # Giant Swarm managed Observability Stack of Apps

Reviewer 2: Sentence case, except for "Apps"

```suggestion
# Giant Swarm managed observability stack of Apps
```

Observability is based on three main data sources: logs, metrics and tracing\*. To cover these needs Giant Swarm provides its customers with a managed observability stack. The apps included in this stack are listed below. We chose these apps based on our general principles of preferring open source over proprietary solutions, as well as simplicity, performance, and speed over complexity.

**What do we mean by a "managed" app?** We mean offering similar Service Level Objectives as the rest of our offering as described in [The Radical Way Giant Swarm Handles Service Level Objectives](https://www.giantswarm.io/blog/the-radical-way-giant-swarm-handles-service-level-objectives).

In brief, super fast response times from engineers, 24//7 monitoring and alerting by us, and in general, having shared goals and responsibilities to keep services up and get them up when they go down.

In addition, we strive to uphold our "30 Day Upgrade Promise," our commitment to keep all managed apps within 30 days of latest upstream releases.

> TL;DR
> For observability, we recommend and provide these four managed apps:

Reviewer 2: ```suggestion
To summarize, for observability, we recommend and provide these four managed apps:
```

> ## 1/ Prometheus for monitoring / metrics

Reviewer 2: The slash after the number is strange. But I don't think a number is required here at all.

```suggestion
## Prometheus for monitoring/metrics
```

To get all your metrics, process, store and alert on them. Prometheus has become the de-facto standard monitoring tool in the kubernetes ecosystem.

> ## 2/ Loki for logging

Reviewer 2: ```suggestion
## Loki for logging
```

To fetch logs from containers, store and index them, and make them searchable. To install, go to [Loki README](https://github.com/giantswarm/loki-app/blob/master/README.md).

In addition, we understand some customers already know and use Elasticsearch. For this reason, we also provide a managed EFK stack (Elasticsearch, Fluentd, and Kibana) for logging. To install, go to [EFK Stack README](https://github.com/giantswarm/efk-stack-app/blob/master/README.md).

> ## 3/ Linkerd2 for service mesh

Reviewer 2: ```suggestion
## Linkerd2 for service mesh
```

To get request metrics and for service discovery, as well as automatic mTLS encryption between all involved services. To install, go to [Linkerd 2 Quickstart Guide](https://github.com/giantswarm/linkerd2-app/blob/master/README.md).

> ## 4/ Grafana as data visualization tool

Reviewer 2: ```suggestion
## Grafana as data visualization tool
```

To analyze, visualize, and correlate metrics from Prometheus and Linkerd as well as logs from Loki.

These pieces can be separately or together. Adopting them together and as fully managed apps from Giant Swarm gives you the benefit of using vetted open source tools that you know work together to provide you with consistency and coverage.

> *We are currently not offering a fully managed app for tracing due to the low priority it has with our customers. We believe that at this time the [tracing options provided by Linkerd](https://www.giantswarm.io/blog/part-5-traces-of-your-microservices-0) (that require no code changes) are sufficient.

Reviewer 2: ```suggestion
We are currently not offering a fully managed app for tracing due to the low priority it has with our customers. We believe that at this time the [tracing options provided by Linkerd](https://www.giantswarm.io/blog/part-5-traces-of-your-microservices-0) (that require no code changes) are sufficient.
```
